src/main/resources/interface-specification.yml

```diff
@@ -76,18 +70,15 @@ paths:
       parameters:
         - in: query
           name: search
-          required: true
           schema:
             type: string
         - in: query
           name: page
```

Comment:
suggest adding a `min` constraint to let the caller know that the first page is 1.
Maybe having the default value here would be nice (same for other `page` and `limit` parameters)

Reply:
I put them, but the generator treated them as required fields without putting any constraints or default value.

This is without constraints:
```
def getCategories(origin: Option[String], page: Option[Int], limit: Option[Int])
```
and this with openapi constraints set:
```
def getCategories(origin: Option[String], page: Int, limit: Int)
```
On the client generation side this is the generated code with the constraints:
```
def getCategories(origin: Option[String] = None, page: Option[Int] = None, limit: Option[Int] = None)
```

I agree that putting them is very useful on the API side, but generator does work as expected.
A solution is to restore these fields as mandatory and check their limitation directly in the implementation.

Reply:
I was missing where exactly the generator passes the defaults, so my previous comment is invalid.